packages/app/src/cli/services/context/identifiers.ts

```diff
@@ -36,8 +36,9 @@ export interface LocalSource {
 export type MatchingError = 'pending-remote' | 'invalid-environment' | 'user-cancelled'
 
 export async function ensureDeploymentIdsPresence(options: EnsureDeploymentIdsPresenceOptions) {
-  // We need local extensions to deploy
-  if (!options.app.hasExtensions()) return {app: options.appId, extensions: {}, extensionIds: {}}
+  // We need local extensions to deploy when unifiedAppDeployment is disabled
+  if (!options.app.hasExtensions() && !options.partnersApp?.betas?.unifiedAppDeployment)
```

Comment:
It looks like the changes are made off of the `main` branch. But this code is already going to change as part of `deployments-for-3-48` branch and the code you wrote for the `app deploy` command. I wonder if instead we could create a feature branch off of `deployments-for-3-48` and instead of merging with `main` merge it into `deployments-for-3-48`. This way we can keep `deployments-for-3-48` branch up to date with all deployment related changes.

Reply:
Okay, I'll merge this into our `deployments-for-3-48` branch